spark/src/main/scala/io/delta/tables/DeltaMergeBuilder.scala

```diff
@@ -265,14 +265,6 @@ class DeltaMergeBuilder private(
    */
   def execute(): Unit = improveUnsupportedOpError {
     val sparkSession = targetTable.toDF.sparkSession
-    // Note: We are explicitly resolving DeltaMergeInto plan rather than going to through the
-    // Analyzer using `Dataset.ofRows()` because the Analyzer incorrectly resolves all
-    // references in the DeltaMergeInto using both source and target child plans, even before
-    // DeltaAnalysis rule kicks in. This is because the Analyzer  understands only MergeIntoTable,
-    // and handles that separately by skipping resolution (for Delta) and letting the
-    // DeltaAnalysis rule do the resolving correctly. This can be solved by generating
-    // MergeIntoTable instead, which blocked by the different issue with MergeIntoTable as explained
-    // in the function `mergePlan` and https://issues.apache.org/jira/browse/SPARK-34962.
```

Comment:
looks like ``` toDataset(sparkSession, mergePlan)``` works now (only executed  MergeIntoSQLSuite )
I'm not sure if it was working before or not

Reply:
`MergeIntoSQLSuite` isn't impacted by this change - it only uses the SQL API, the suite to run is `MergeIntoScalaSuite`
I gave it a try and multiple tests fail: the `DeltaMergeInto` plan gets resolved by spark generic resolution rule instead of `DeltaAnalysis` which screws resolution.

Assuming https://issues.apache.org/jira/browse/SPARK-34962 is correctly addressed, the path forward would be to create a `MergeIntoTable` plan instead of `DeltaMergeInto` and things should then (hopefully) just work by just calling `toDataset(sparkSession, mergePlan)`. That's a pretty intrusive change though, I won't look into that here